parsers/CZ.py

```diff
@@ -68,6 +68,71 @@ def get_target_datetime(dt: datetime | None) -> datetime:
     return dt
 
 
+def fetch_consumption(
+    zone_key: str = "CZ",
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> List[dict]:
+
+    target_datetime = get_target_datetime(target_datetime)
+    from_datetime = target_datetime - timedelta(hours=48)
+
+    payload = """<?xml version="1.0" encoding="utf-8"?>
+    <soap12:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap12="http://www.w3.org/2003/05/soap-envelope">
+      <soap12:Body>
+        <Load xmlns="https://www.ceps.cz/CepsData/">
+          <dateFrom>{0}</dateFrom>
+          <dateTo>{1}</dateTo>
+          <agregation>{2}</agregation>
+          <function>{3}</function>
+          <version>{4}</version>
+        </Load>
+      </soap12:Body>
+    </soap12:Envelope>""".format(
+        from_datetime.isoformat(), target_datetime.isoformat(), "QH", "AVG", "RT"
+    )
+
+    content = make_request(session, payload, zone_key).text
+    xml = BeautifulSoup(content, "xml")
+    mapper = get_mapper(xml)
+
+    data_tag = xml.find("data")
+    data_list: List[dict] = []
+
+    if data_tag is not None:
+        for values in data_tag:
```

Comment:
This whole looping over xml elements is filled with type errors. Not sure we can merge it as is.